lib/ldp.js

```diff
@@ -156,42 +154,40 @@ class LDP {
       return callback(error(500, "Can't parse container"))
     }
 
-    async.waterfall(
-      [
-        // add container stats
-        function (next) {
-          ldpContainer.addContainerStats(ldp, reqUri, filename, resourceGraph, next)
-        },
-        // reading directory
-        function (next) {
-          ldpContainer.readdir(filename, next)
-        },
-        // Iterate through all the files
-        function (files, next) {
-          async.each(
-            files,
-            function (file, cb) {
-              let fileUri = url.resolve(reqUri, encodeURIComponent(file))
-              ldpContainer.addFile(ldp, resourceGraph, reqUri, fileUri, uri,
-                filename, file, cb)
-            },
-            next)
-        }
-      ],
-      function (err, data) {
+    // add container stats
+    new Promise((resolve, reject) =>
+      ldpContainer.addContainerStats(ldp, reqUri, filename, resourceGraph,
+        err => err ? reject(err) : resolve())
```

Comment:
[Won't have been me](https://github.com/RubenVerborgh/N3.js/issues/111) 😄  I'm in favor of a [strict .eslintrc](https://github.com/RubenVerborgh/N3.js/blob/master/.eslintrc) (and really dislike `standard` :wink:).